contracts/src/Summa.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.18;
+
+// Uncomment this line to use console.log
+//import "hardhat/console.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface Verifier {
+    function verify(
+        uint256[] calldata pubInputs,
+        bytes calldata proof
+    ) external view returns (bool);
+}
+
+contract Summa is Ownable {
+    using ECDSA for bytes32;
+
+    uint256 constant ETH_INPUTS = 1;
```

Comment:
Is there a way to not hardcode ETH_INPUTS and ERC20_INPUTS?

Reply:
If we don't hard-code them, we'll have to hard-code the input balance array size. The verifier always expects an input array of a known size so I don't think it's a problem to have that number explicitly defined.

Reply:
I see now! Actually the lenght of the array `inputs` that is taken by the verifier is not depended on `ETH_INPUTS` and `ERC20_INPUTS`. 

For example, let's say that a CEX has 20 wallets with eth on them. The wallet provides all length-20-arrays `cexETHAddresses`, `cexETHBalances` and `cexEthAddressSignatures` to the `submitProofOfSolvency` function. The public input of the circuit will be `cexETHAssetsSum` which must be equal to ƩcexETHBalances.